app/data_model/app_models.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+
+from dateutil.tz import tzutc
+from marshmallow import Schema, fields, post_load, pre_dump
+
+
+class QuestionnaireState:
+    def __init__(self, user_id, state_data, version, created_at=None, updated_at=None):
+        self.user_id = user_id
+        self.state_data = state_data
+        self.version = version
+        self.created_at = created_at or datetime.datetime.now(tz=tzutc())
+        self.updated_at = updated_at or datetime.datetime.now(tz=tzutc())
+
+
+class EQSession:
+    def __init__(self, eq_session_id, user_id, session_data=None, created_at=None, updated_at=None):
+        self.eq_session_id = eq_session_id
+        self.user_id = user_id
+        self.session_data = session_data
+        self.created_at = created_at or datetime.datetime.now(tz=tzutc())
+        self.updated_at = updated_at or datetime.datetime.now(tz=tzutc())
+
+
+class UsedJtiClaim:
+    def __init__(self, jti_claim, used_at, valid_until):
+        self.jti_claim = jti_claim
+        self.used_at = used_at
+        self.valid_until = valid_until
+
+
+class SubmittedResponse:
+    def __init__(self, tx_id, data, valid_until):
+        self.tx_id = tx_id
+        self.data = data
+        self.valid_until = valid_until
+
+
+# pylint: disable=no-self-use
+class Timestamp(fields.Field):
+    def _serialize(self, value, attr, obj):
+        return int(value.strftime('%s'))
+
+    def _deserialize(self, value, attr, data):
+        return datetime.datetime.utcfromtimestamp(value).replace(tzinfo=tzutc())
+
+
+class DateTimeSchemaMixin:
+    created_at = fields.DateTime()
+    updated_at = fields.DateTime()
+
+    @pre_dump
+    def set_date(self, data):
+        data.updated_at = datetime.datetime.now(tz=tzutc())
+        return data
+
+
+class QuestionnaireStateSchema(Schema, DateTimeSchemaMixin):
+    user_id = fields.Str()
+    state_data = fields.Str()
+    version = fields.Integer()
+
+    @post_load
+    def make_model(self, data):
+        return QuestionnaireState(**data)
+
+
+class EQSessionSchema(Schema, DateTimeSchemaMixin):
+    eq_session_id = fields.Str()
+    user_id = fields.Str()
+    session_data = fields.Str()
+
+    @post_load
+    def make_model(self, data):
+        return EQSession(**data)
+
+
+class UsedJtiClaimSchema(Schema):
+    jti_claim = fields.Str()
+    used_at = fields.DateTime()
+    valid_until = Timestamp()
```

Comment:
This technically isn't `valid_until` its more like `invalid_until` as if its in our DB then it can't be used